### PR TITLE
Add script that evaluates mesh model on our grants

### PR DESCRIPTION
**Pull request by nsorros**
This script evaluates the current mesh model performance (tfidf-svm) in our grants data. To do that, Kirstin mapped the ICD codes that the annotators had used to MeSH.

The data collated the model predictions with the gold annotations, in the future we would like to use the model to produce the predictions so that we evaluate different models.

### CODE REVIEW

File: scripts/evaluate_mesh_on_grants.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    data = pd.read_excel(data_path)
+
+    with open(label_binarizer_path, "rb") as f:
+        label_binarizer = pickle.loads(f.read())
+
+    gold_tags = get_tags(data, "KW")
+    model_tags = get_tags(data, "NS")
+    Y = label_binarizer.transform(gold_tags)
+    Y_pred = label_binarizer.transform(model_tags)
+#    print(classification_report(Y, Y_pred, target_names=label_binarizer.classes_))
```
Comment (nsorros) [inline scripts/evaluate_mesh_on_grants.py:34]: remove comment

**Comment by nsorros** [comment]
[Evaluate disease mesh tagger on a sample of our grants tagged from annotators by mapping ICD codes to Mesh codes](https://trello.com/c/qSyVVCsb/1096-evaluate-disease-mesh-tagger-on-a-sample-of-our-grants-tagged-from-annotators-by-mapping-icd-codes-to-mesh-codes)

File: scripts/evaluate_mesh_on_grants.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""
+Evaluate MeSH model on grants annotated with MeSH
+
+The script works on an Excel file that contains 
+columns that follow the pattern below
+Disease MeSH#TAG_INDEX (ANNOTATOR)
```
Comment (aCampello) [inline scripts/evaluate_mesh_on_grants.py:6]: Did Kirstin do all this manually? Impressive.

File: scripts/evaluate_mesh_on_grants.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    return tags
+
+def evaluate_mesh_on_grants(data_path, label_binarizer_path):
+    data = pd.read_excel(data_path)
+
+    with open(label_binarizer_path, "rb") as f:
+        label_binarizer = pickle.loads(f.read())
+
+    gold_tags = get_tags(data, "KW")
+    model_tags = get_tags(data, "NS")
```
Comment (aCampello) [inline scripts/evaluate_mesh_on_grants.py:31]: Nice aliases... 😉

**Review by aCampello — APPROVED** [review]
Looks good. What is the result? Maybe update the documentation with the current results?

File: scripts/evaluate_mesh_on_grants.py (hunk 2 above)
Comment (nsorros) [reply scripts/evaluate_mesh_on_grants.py:6]: This is 2000 grants and she mapped previous annotators but 100% still impressive. Shout to @kirstinwilliamson 🥇

File: scripts/evaluate_mesh_on_grants.py (hunk 3 above)
Comment (nsorros) [reply scripts/evaluate_mesh_on_grants.py:31]: Again shout out ot @kirstinwilliamson

**Comment by nsorros** [comment]
> Looks good. What is the result? Maybe update the documentation with the current results?

Good point, let me see where best to add it. The result was 69% micro f1 which is impressive since the model performed 56% on the test set but it is partly explained by the non blind process i.e. Kirstin was looking at the model predictions and sometimes agreeing

**Pull request merged**